core/app/[locale]/(default)/(auth)/change-password/page.tsx

```diff
@@ -1,31 +1,37 @@
 /* eslint-disable react/jsx-no-bind */
 import { Metadata } from 'next';
-import { getLocale, getTranslations } from 'next-intl/server';
+import { getTranslations, setRequestLocale } from 'next-intl/server';
 
 import { ResetPasswordSection } from '@/vibes/soul/sections/reset-password-section';
 import { redirect } from '~/i18n/routing';
 
 import { changePassword } from './_actions/change-password';
 
 interface Props {
+  params: Promise<{ locale: string }>;
   searchParams: Promise<{
     c?: string;
     t?: string;
   }>;
 }
 
-export async function generateMetadata(): Promise<Metadata> {
-  const t = await getTranslations('ChangePassword');
+export async function generateMetadata({ params }: Props): Promise<Metadata> {
+  const { locale } = await params;
+
+  const t = await getTranslations({ locale, namespace: 'ChangePassword' });
 
   return {
     title: t('title'),
   };
 }
 
-export default async function ChangePassword({ searchParams }: Props) {
+export default async function ChangePassword({ params, searchParams }: Props) {
+  const { locale } = await params;
+
+  setRequestLocale(locale);
+
   const { c: customerEntityId, t: token } = await searchParams;
   const t = await getTranslations('ChangePassword');
-  const locale = await getLocale();
```

Comment:
This was a weird case where we were using `getLocale` instead of `params.locale`. I looked at the history and it seems that @jorgemoya you added `useLocale` a while ago before this was an `async` component and then @chanceaclark you changed it from `useLocale` to `getLocale` after moving to Next.js v15 and having to turn it into an `async` function due to having to await `props.searchParams`.

Please let me know if there's a reason we should continue to use `getLocale` over `params.locale` 🙏🏻